reflex/app.py

```diff
@@ -659,81 +654,128 @@ def compile(self):
 
-        # Iterate through all the custom components and add their imports to the all_imports
-        for component in custom_components:
-            all_imports.update(component.get_imports())
+            # Compile the root document.
+            submit_work(compiler.compile_document_root, self.head_components)
 
-        # Compile the root stylesheet with base styles.
-        compile_results.append(compiler.compile_root_stylesheet(self.stylesheets))
+            # Compile the theme.
+            submit_work(compiler.compile_theme, style=self.style)
 
-        # Compile the root document.
-        compile_results.append(compiler.compile_document_root(self.head_components))
+            # Compile the contexts.
+            submit_work(compiler.compile_contexts, self.state)
 
-        # Compile the theme.
-        compile_results.append(compiler.compile_theme(style=self.style))
+            # Compile the Tailwind config.
+            if config.tailwind is not None:
+                config.tailwind["content"] = config.tailwind.get(
+                    "content", constants.Tailwind.CONTENT
+                )
+            submit_work(compiler.compile_tailwind, config.tailwind)
 
-        # Compile the contexts.
-        compile_results.append(compiler.compile_contexts(self.state))
+            # Get imports from AppWrap components.
+            all_imports.update(app_root.get_imports())
 
-        # Compile the Tailwind config.
-        if config.tailwind is not None:
-            config.tailwind["content"] = config.tailwind.get(
-                "content", constants.Tailwind.CONTENT
-            )
-            compile_results.append(compiler.compile_tailwind(config.tailwind))
+            # Iterate through all the custom components and add their imports to the all_imports.
+            for component in custom_components:
+                all_imports.update(component.get_imports())
 
-        # Empty the .web pages directory
-        compiler.purge_web_pages_dir()
+            # Empty the .web pages directory.
+            compiler.purge_web_pages_dir()
 
-        # install frontend packages
-        self.get_frontend_packages(all_imports)
+            # Install frontend packages.
+            self.get_frontend_packages(all_imports)
 
-        # Write the pages at the end to trigger the NextJS hot reload only once.
-        thread_pool = ThreadPool()
-        for output_path, code in compile_results:
-            thread_pool.apply_async(compiler_utils.write_page, args=(output_path, code))
-        thread_pool.close()
-        thread_pool.join()
+            # Wait for all compilation tasks to complete.
+            for future in concurrent.futures.as_completed(result_futures):
+                compile_results.append(future.result())
+
+            # Write the pages at the end to trigger the NextJS hot reload only once.
+            progress.update(task, description="Write:")
```

Comment:
This after the `get_frontend_packages` cause the progress back to flicker weirdly for me, I think we should get rid of this